src/core/server/saved_objects/saved_objects_service.ts

```diff
@@ -485,4 +492,60 @@ export class SavedObjectsService
       ),
     });
   }
+
+  private selectRepository(config: SavedObjectConfig) {
+    const savedObjectsRepository = OpensearchSavedObjectsRepository;
+    switch (config.dbType) {
+      case 'opensearch':
+        return savedObjectsRepository;
+      case 'postgres':
+        this.createTable(config);
+        return PostgresSavedObjectsRepository;
+      default:
+        return savedObjectsRepository;
+    }
+  }
+
+  private async createTable(config: SavedObjectConfig) {
```

Comment:
I am just curious what data will be saved in this MetaDataStore. Can you please share the sample data entry in the Postgres SQL DB after the index migration?

Reply:
idtext | typetext | versiontext | attributesjsonb | referencejsonb | migrationversionjsonb | namespacestext[] | originidtext | updated_attext
-- | -- | -- | -- | -- | -- | -- | -- | --
search:571aaf70-4c88-11e8-b3d7-01146121b73d | search |   | {"type":"search","search":{"hits":0,"sort":[["timestamp","desc"]],"title":"[Flights] Flight Log","columns":["Carrier","OriginCityName","OriginCountry","DestCityName","DestCountry","FlightTimeMin","AvgTicketPrice","Cancelled","FlightDelayType"],"version":1,"description":"","kibanaSavedObjectMeta":{"searchSourceJSON":"{\"highlightAll\":true,\"version\":true,\"query\":{\"language\":\"kuery\",\"query\":\"\"},\"filter\":[],\"indexRefName\":\"kibanaSavedObjectMeta.searchSourceJSON.index\"}"}},"references":[{"id":"d3d7af60-4c81-11e8-b3d7-01146121b73d","name":"kibanaSavedObjectMeta.searchSourceJSON.index","type":"index-pattern"}],"updated_at":"2023-02-24T18:36:35.040Z","migrationVersion":{"search":"7.9.3"}} | [{"id":"d3d7af60-4c81-11e8-b3d7-01146121b73d","name":"kibanaSavedObjectMeta.searchSourceJSON.index","type":"index-pattern"}] | {"search":"7.9.3"} |   |   | 2023-02-24T18:36:35.040Z